design/how-ironic-works.md

```diff
@@ -146,6 +146,34 @@ Between ironic and ironic-inspector:
 * Each service must be able to reach the API endpoint for
   the the other service.
 
+## What is ironic-inspector?
+Ironic-inspector is an auxiliary service that provides separated API to
+inspect and register hardware properties for a bare metal node.
+This process may be triggered during discovery and allow automatic node
+registration in the ironic DB.
+The process of "hardware introspection" allows also to get the hardware
+parameters from a bare metal node and ready it for scheduling.
+Power management credentials should be already set to allow node
+introspection.
+
+Ironic-inspector relies on plugins for data processing even for standard
+functionality; basic hardware introspection plugins are enabled by
+default.
```

Comment:
Do we need these two paragraphs here? Seems a bit too detailed.

Reply:
Maybe you're right, I'll stick to an overview of the service without too many under the hood details that can be retrieved from the official docs.
We can always add this to a different section later.